module/bdev/ocf/volume.c

```diff
@@ -235,7 +235,16 @@ prepare_submit(struct ocf_io *io)
 		return -EFAULT;
 	}
 
-	if (q == cctx->cleaner_queue || q == cctx->mngt_queue) {
+	if (q == cctx->cleaner_queue) {
+		if (base->is_cache) {
+			io_ctx->ch = cctx->cleaner_cache_channel;
+		} else {
+			io_ctx->ch = cctx->cleaner_core_channel;
+		}
+		return 0;
+	}
```

Comment:
It was broken before, so this patch doesn't make it worse, but it will fail miserably in multi-core configuration.
Note to future ourselves: Fix this on SPDK master.